tutorials/dataframe/tdf012_InspectAnalysis.C

```diff
@@ -0,0 +1,119 @@
+   auto browserPad = gPad;
+
+   // A useful feature of `TResultProxy` is its `OnPartialResult` method: it allows us to register a callback that is
+   // executed once per specified number of events during the event-loop, on "partial" versions of the result objects
+   // contained in the `TResultProxy`. In this case, the partial result is going to be a histogram filled with an
+   // increasing number of events.
+   // Instead of requesting the callback to be executed every N entries, this time we use the special value `kOnce` to
+   // request that it is executed once right before starting the event-loop.
+   // The callback is a C++11 lambda that registers the partial result object in `tdfDirectory`.
+   h.OnPartialResult(h.kOnce, [&tdfDirectory](TH1D &h_) { tdfDirectory.Add(&h_); });
+   // Note that we called `OnPartialResult` with a dot, `.`, since this is a method of `TResultProxy` itself.
+   // We do not want to call `OnPartialResult` on the pointee histogram!)
+
+   // Multiple callbacks can be registered on the same `TResultProxy` (they are executed one after the other in the
+   // same order as they were registered). We now request that the partial result is drawn and the TBrowser's TPad is
+   // updated every 50 events.
+   h.OnPartialResult(50, [&browserPad](TH1D &hist) {
+      browserPad->cd();
+      hist.Draw();
+      browserPad->Update();
+      // This call tells ROOT to process all pending GUI events
+      // It allows users to use the TBrowser as usual while the event-loop is running
+      gSystem->ProcessEvents();
+   });
+
+   // Finally, we would like to print a progress bar on the terminal to show how the event-loop is progressing.
+   // To take into account _all_ events we use `OnPartialResultSlot`: when Implicit Multi-Threading is enabled, in fact,
+   // `OnPartialResult` invokes the callback only in one of the worker threads, and always returns that worker threads'
+   // partial result. This is useful because it means we don't have to worry about concurrent execution and
+   // thread-safety of the callbacks if we are happy with just one threads' partial result.
+   // `OnPartialResultSlot`, on the other hand, invokes the callback in each one of the worker threads, every time a
+   // thread finishes processing a batch of `everyN` events. This is what we want for the progress bar, but we need to
+   // take care that two threads will not print to terminal at the same time: we need a std::mutex for synchronization.
+   std::string progressBar;
+   std::mutex barMutex; // Only one thread at a time can lock a mutex. Let's use this to avoid concurrent printing.
+   // Magic numbers that yield good progress bars for nSlots = 1,2,4,8
+   const auto everyN = nSlots == 8 ? 1000 : 100ull * nSlots;
+   const auto barWidth = nEvents / everyN;
+   h.OnPartialResultSlot(everyN, [&barWidth, &progressBar, &barMutex](unsigned int, TH1D &) {
```

Comment:
Since this is a tutorials, maybe add a hint on what the purpose of the first parameter is:
```
(unsigned int /* semanticName */, TH1D &) 
```

Reply:
I thought `everyN` was semantic enough, what about `everyNEvents`?

Reply:
> I thought everyN was semantic enough, what about everyNEvents?

I meant the first argument of the lambda function :).

Reply:
Oh sorry, sure, will do